github.go

```diff
@@ -11,10 +11,12 @@ import (
 )
 
 type GitHubRepo struct {
-	Url   string // The URL of the GitHub repo
-	Owner string // The GitHub account name under which the repo exists
-	Name  string // The GitHub repo name
-	Token string // The personal access token to access this repo (if it's a private repo)
+	Url     string // The URL of the GitHub repo
+	BaseUrl string // The Base URL of the GitHub Instance
+	ApiUrl  string // The API Url of the GitHub Instance
```

Comment:
Are these 3 ever different? Or, more accurately, is it always the case where:

```go
Url = <domain>/<asset_path>
BaseUrl = <domain>
ApiUrl = <domain>/api/<version>
```

I'm mainly trying to figure out if we could deduce `<domain>` automatically from `Url` without a new parameter and have an optional `ApiVersion` param that defaults to something reasonable (`v1`)...

Reply:
On github.com, the ApiUrl is always `api.github.com` and the version can be chosen using an HTTP header. For [enterprise][ghe], it's always `baseurl.com/api/v3`.

So we probably could use the repo url to do something like:

```
baseUrl := parseBaseFromRepoUrl(options.RepoUrl)
apiUrl := "api.github.com"

if baseUrl != "github.com" {
  apiUrl := baseUrl+"/api/"+options.ApiVersion
}
```

[ghe]: https://developer.github.com/enterprise/2.14/v3/enterprise-admin/
[gh]: https://developer.github.com/v3/

Reply:
That seems reasonable to me and, if `ApiVersion` is set to a default of `v3`, it makes the user experience easier, as things will "just work" with no extra parameters. Might want to add a logging statement in that if-statement to make it explicit a non github.com URL was found. Also, does that if-statement also need to check against `www.github.com`?

Reply:
Could I get temporary read access to: `https://github.com/gruntwork-io/fetch-test-private`. So that some of the other acceptance tests pass?

Reply:
@brikis98 I made the changes you suggested and updated the test suite. This is ready for review.

Reply:
Ah, I always forget about that. We have to have a private repo for testing, but then, of course, contributors can't access it 😄 

Just added you.